duckduckgo/chrome-zeroclickinfo
# Russell/trackerwhitelist

This is a way to whitelist trackers separately from existing lists. These will still count against the Privacy Grade, like trackers are required to operate a site.

This one includes a rule for the 2mdn.net video player – probably applied too broadly at the moment.

It uses the easylist parser/matcher abp.js.
Normally, abp list items are meant to be 'blocked', that is, an easylist whitelist entry means "don't block it", ie, not a tracker. Easy list, used in the general way, won't distinguish between tracker to be blocked, and tracker to not be blocked.

In the case of 2mdn.net, this domain is in our parent company tracker list (owned by Google), which is checked before our easy lists; removing that or changing the way we use that list to unblock trackers seems like the wrong way to do it.

Also, switching the order to whitelist 2mdn.net via adblock whitelists also seems wrong, since that would again mark this as not a tracker.

Therefore I introduced a new list to be checked outside this "is tracker / is not a tracker" checking, right at the point of the block decision. It's in normal adblock format, in data/trackerWhitelist.txt. But not *whitelist* format, as that would return a negative match (ie not a tracker).


## Steps to test this PR:

Go to http://www.clevelandbrowns.com/media-center/videos/FanDuel-Fantasy-Minute---2017-Week-2/28303837-0891-4b5e-ada8-3842b315910e

- On the previous version, the video will never load.
- With this version, it will autoplay.

Look for 'UNBLOCKED' in the background page inspector, you should see:

```
UNBLOCKED clevelandbrowns.com [Google] http://s0.2mdn.net/instream/html5/ima3.js
```

## Next steps:

1. Decide if this method is good in general.
2. Decide if this particular rule is too broad. The easyprivacy lists specifies some specific domains to whitelist for this.
3. Consider moving the whitelist to the contentblocking.js endpoint


## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
